akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -253,6 +251,8 @@ import akka.util.unused
           if (!receivedInput && isAllCompleted()) onCompleteWhenNoInput()
         case OnError(id, e) =>
           onError(id, e)
+        case OnSubscribe(id, subscription) =>
```

Comment:
OnNext is absolutely most common, others only happens once. I would place this before OnComplete, just to preserve the logical order that it happens before OnComplete